agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/quickpulse/QuickPulsePingSender.java

```diff
@@ -93,10 +101,11 @@ public QuickPulseHeaderInfo ping(String redirectedEndpoint) {
             machineName,
             telemetryClient.getRoleName(),
             instanceName);
-    request.setBody(buildPingEntity(currentDate.getTime()));
 
     long sendTime = System.nanoTime();
-    try (HttpResponse response = httpPipeline.send(request).block()) {
+    try {
+      request.setBody(buildPingEntity(currentDate.getTime()));
+      HttpResponse response = httpPipeline.send(request).block();
```

Comment:
need to close the response in finally block now that it's not using try-with-resources